sim/battle-actions.ts

```diff
@@ -1635,6 +1636,8 @@ export class BattleActions {
 		// int(int(int(2 * L / 5 + 2) * A * P / D) / 50);
 		const baseDamage = tr(tr(tr(tr(2 * level / 5 + 2) * basePower * attack) / defense) / 50);
 
+		if (baseDamage > 0) target.hitSelf = false;
```

Comment:
This isn't an indicator of the Pokemon took damage right? Base damage isn't the same as the actual damage roll. Moreover, if an attack didn't use base damage (say Seismic Toss) wouldn't this check get bypassed?